src/main/java/org/nibor/autolink/LinkExtractor.java

```diff
@@ -27,11 +27,12 @@ public static Builder builder() {
 
     /**
      * Extract the links from the input text. Can be called multiple times with different inputs (thread-safe).
-     *
-     * @param input the input text, must not be {@code null}
+     * @param input the input text
      * @return a lazy iterable for the links in order that they appear in the input, never {@code null}
+     * @throws NullPointerException if the input argument is null 
      */
     public Iterable<LinkSpan> extractLinks(final CharSequence input) {
+    	Objects.requireNonNull(input);
```

Comment:
Could you add a message as a second argument, like this?:

    Objects.requireNonNull(input, "input must not be null");